### PR TITLE
Add Zeppelin

### LINKED ISSUE #39: Add Zeppelin

### PR DESCRIPTION
**Pull request by pan3793**
Close #39 

```
ansible-playbook playbook.yaml -e "{zeppelin_enabled: true}"
./download.sh
./build-image.sh
docker compose up
```

http://hadoop-master1.orb.local:8081

image

image

### CODE REVIEW

File: templates/hadoop-master/files/etc/zeppelin/conf/zeppelin-site.xml [hunk 1]
```diff
@@ -0,0 +1,826 @@
...
+ </description>
+</property>
+
+
+<property>
+  <name>zeppelin.notebook.storage</name>
+  <value>org.apache.zeppelin.notebook.repo.GCSNotebookRepo</value>
+  <description>notebook persistence layer implementation</description>
+</property>
+-->
```
Comment (yanghua) [inline templates/hadoop-master/files/etc/zeppelin/conf/zeppelin-site.xml:125]: Can we delete these useless configuration items?

File: templates/hadoop-master/files/etc/zeppelin/conf/zeppelin-site.xml [hunk 2]
```diff
@@ -0,0 +1,826 @@
...
+<property>
+  <name>zeppelin.notebook.s3.timeout</name>
+  <value>120000</value>
+  <description>s3 bucket endpoint request timeout in msec.</description>
+</property>
+
+<property>
+  <name>zeppelin.notebook.storage</name>
+  <value>org.apache.zeppelin.notebook.repo.S3NotebookRepo</value>
+  <description>notebook persistence layer implementation</description>
```
Comment (yanghua) [inline templates/hadoop-master/files/etc/zeppelin/conf/zeppelin-site.xml:157]: ditto

File: templates/hadoop-master/files/etc/zeppelin/conf/zeppelin-site.xml [hunk 3]
```diff
@@ -0,0 +1,826 @@
...
+<!-- Use the AWS KMS to encrypt data -->
+<!-- If used, the EC2 role assigned to the EMR cluster must have rights to use the given key -->
+<!-- See https://aws.amazon.com/kms/ and http://docs.aws.amazon.com/kms/latest/developerguide/concepts.html -->
+<!--
+<property>
+  <name>zeppelin.notebook.s3.kmsKeyID</name>
+  <value>AWS-KMS-Key-UUID</value>
+  <description>AWS KMS key ID used to encrypt notebook data in S3</description>
+</property>
+-->
```
Comment (yanghua) [inline templates/hadoop-master/files/etc/zeppelin/conf/zeppelin-site.xml:172]: ditto

**Review by yanghua — COMMENTED** [review]
Left some comments. It seems `zeppelin-site.xml` has many useless configurations.

File: templates/base-ubuntu-2004/Dockerfile.j2 [hunk 4]
```diff
@@ -24,8 +24,9 @@ RUN cat /etc/apt/sources.list.mirror-$(uname -m) > /etc/apt/sources.list
 
 RUN set -xeu && \
     ln -snf /usr/bin/bash /usr/bin/sh && \
-    install_packages busybox python3-pip supervisor xsltproc curl tree python-is-python3 openssh-client openssh-server \
-    krb5-user jsvc libssl1.1 && \
+    install_packages busybox python3-pip supervisor xsltproc curl tree python-is-python3 \
+    openssh-client openssh-server sqlite3 \
```
Comment (yanghua) [inline templates/base-ubuntu-2004/Dockerfile.j2:28]: Where is the use case for `sqlite3`?

**Comment by pan3793** [comment]
let me separate the unrelated changes into dedicated PRs

File: templates/hadoop-master/files/etc/zeppelin/conf/zeppelin-site.xml (hunk 2 above)
Comment (pan3793) [reply templates/hadoop-master/files/etc/zeppelin/conf/zeppelin-site.xml]: done

File: templates/hadoop-master/files/etc/zeppelin/conf/zeppelin-site.xml (hunk 3 above)
Comment (pan3793) [reply templates/hadoop-master/files/etc/zeppelin/conf/zeppelin-site.xml]: done

File: templates/base-ubuntu-2004/Dockerfile.j2 (hunk 4 above)
Comment (pan3793) [reply templates/base-ubuntu-2004/Dockerfile.j2]: submitted in https://github.com/awesome-kyuubi/hadoop-testing/pull/87

**Comment by pan3793** [comment]
let me merge it first, and do optimization later if needed

**Pull request merged**